networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/Eth2GossipRouter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.p2p.libp2p;
+
+import io.libp2p.pubsub.gossip.GossipRouter;
+import org.apache.tuweni.crypto.Hash;
+import org.apache.tuweni.io.Base64;
+import org.jetbrains.annotations.NotNull;
+import pubsub.pb.Rpc.Message;
+
+/** Customization of a standard Libp2p Gossip router for Eth2 spec */
+public class Eth2GossipRouter extends GossipRouter {
+
+  @NotNull
+  @Override
+  protected String getMessageId(@NotNull Message msg) {
+    return Base64.encodeBytes(Hash.sha2_256(msg.getData().toByteArray()));
```

Comment:
I think this is the wrong base64.  It should be the URL safe variant without padding.  To do that in discv5 I had to use: https://github.com/PegaSysEng/discovery/blob/247107e61cd53ed4c05ea2401985094ae1c3668d/src/main/java/org/ethereum/beacon/discovery/schema/NodeRecord.java#L84

Reply:
Oh, thanks! Absolutely my bad, should read the spec more carefully. 
Don't even think there might be 'wrong' base64 :)

Reply:
BTW, wouldn't this work? 
```java
java.util.Base64.getUrlEncoder().withoutPadding()
```

Reply:
I did not know that method existed.... https://github.com/PegaSysEng/discovery/pull/43 will switch discovery to use it.